app/src/main/kotlin/compiler/ast/Expression.kt

```diff
@@ -23,6 +23,24 @@ sealed class Expression : AstNode {
         override val location: LocationRange? = null,
     ) : Expression()
 
+    data class ArrayElement(
+        val expression: Expression,
+        val index: Expression,
+        override val location: LocationRange? = null
+    ) : Expression()
+
+    data class ArrayLength(
+        val expression: Expression,
+        override val location: LocationRange? = null
+    ) : Expression()
+
+    data class ArrayAllocation(
+        val name: String,
+        val size: Int,
```

Comment:
Do I understand correctly that this node is responsible for both (`arr = {2, 4, 3}` and `arr = new int[10]`) array creation methods?

Reply:
Wasn't the syntax supposed to be `arr = new int[3](1, 2, 3)` (for the first case)? I remember we are against allowing default values, but I could be wrong.

Reply:
It was supposed to be `arr = new int[3](1)`. The value 1 is copied 3 times.